### PR TITLE
geogig 1.2.0 (new formula)

### PR DESCRIPTION
**Pull request by walker**
Initial geogig formula for homebrew-core.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by woodruffw** [comment]
```
==> /usr/local/Cellar/geogig/1.2.0/bin/geogig init
Error: JAVA_HOME is not defined correctly.
  We cannot execute /System/Library/Frameworks/JavaVM.framework/Versions/CurrentJDK/Home/bin/java
Error: geogig: failed
Failed executing: /usr/local/Cellar/geogig/1.2.0/bin/geogig init
```

**Comment by DomT4** [comment]
It's possible you want something like [this](https://github.com/Homebrew/homebrew-core/blob/f096888d87dba5ddec6a4d74307308d70c3adb40/Formula/zanata-client.rb#L12-L13) based on @woodruffw's comment above.

**Comment by walker** [comment]
Thanks @DomT4. That helps greatly.

**Comment by walker** [comment]
@woodruffw Learned alot more about this all while I fixed it. Thanks @DomT4 for the pointer.

Let me know if any of it still seems incorrect.

File: Formula/geogig.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  homepage "http://geogig.org"
+  url "http://download.locationtech.org/geogig/geogig-1.2.0.zip"
+  sha256 "c438d97d8978a9d4985bd4f2ba3fff481b7956bfc4bd986ee3c79c6c07cd4f34"
+
+  bottle :unneeded
+
+  depends_on :java => ["1.8+"]
+
+  def install
+    sbin.install "bin/geogig"
```
Comment (DomT4) [inline Formula/geogig.rb]: Don't put the same executable in `sbin` and `bin`. You'll end up with one that behaves correctly and one that borks because it's not inside the Java wrapper.

**Comment by woodruffw** [comment]
> Let me know if any of it still seems incorrect.

I'll give the other maintainers time to give feedback, but it looks good to me. Thanks!

**Comment by walker** [comment]
@DomT4 Where should I place it? I need to move it somewhere so I can place the Java environment variable in the geogig that gets called when used via terminal.

**Comment by DomT4** [comment]
Ideally, `libexec/bin`.

**Comment by walker** [comment]
@DomT4 The locationtech-distributed package won't work when placed in a sub-folder of the libexec.

```
Error: Could not find or load main class org.locationtech.geogig.cli.app.CLI
Caused by: java.lang.ClassNotFoundException: org.locationtech.geogig.cli.app.CLI
```

**Comment by DomT4** [comment]
Normally we'd do something like this:
```diff
diff --git a/Formula/geogig.rb b/Formula/geogig.rb
index 9a6fd42a7b..b2b9472819 100644
--- a/Formula/geogig.rb
+++ b/Formula/geogig.rb
@@ -9,10 +9,10 @@ class Geogig < Formula
   depends_on :java => ["1.8+"]
 
   def install
-    sbin.install "bin/geogig"
-    prefix.install "libexec"
-    prefix.install "misc"
-    (bin/"geogig").write_env_script "#{sbin}/geogig", Language::Java.java_home_env("1.8+")
+    inreplace "bin/geogig", /BASEDIR=.*/, "BASEDIR=#{libexec}"
+    rm_f Dir["bin/*.bat"]
+    libexec.install Dir["*"]
+    (bin/"geogig").write_env_script "#{libexec}/bin/geogig", Language::Java.java_home_env("1.8+")
   end
 
   test do
```

This raises this failure running `brew test`, which I haven't yet quite worked out:
```
==> /usr/local/Cellar/geogig/1.2.0/bin/geogig init
Picked up _JAVA_OPTIONS:  -Duser.home=/usr/local/var/homebrewcache/java_cache
com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) Error injecting constructor, java.lang.ExceptionInInitializerError
  at org.locationtech.geogig.rocksdb.RocksdbObjectDatabase.<init>(Unknown Source)
  while locating org.locationtech.geogig.rocksdb.RocksdbObjectDatabase
  at org.locationtech.geogig.cli.CLIContextBuilder.bind(CLIContextBuilder.java:119) (via modules: com.google.inject.util.Modules$OverrideModule -> org.locationtech.geogig.cli.CLIContextBuilder$DefaultPlugins)
  while locating org.locationtech.geogig.storage.ObjectDatabase annotated with @com.google.inject.multibindings.Element(setName=,uniqueId=22, type=MAPBINDER, keyType=org.locationtech.geogig.storage.VersionedFormat)

1 error
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1025)
	at com.google.inject.spi.ProviderLookup$1.get(ProviderLookup.java:104)
	at com.google.inject.spi.ProviderLookup$1.get(ProviderLookup.java:104)
	at com.google.inject.multibindings.MapBinder$RealMapBinder$ValueProvider.get(MapBinder.java:746)
	at org.locationtech.geogig.di.FormatSelector.get(FormatSelector.java:41)
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81)
	at com.google.inject.internal.BoundProviderFactory.provision(BoundProviderFactory.java:72)
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:61)
	at com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:62)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:145)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1016)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1012)
	at org.locationtech.geogig.di.GuiceContext.getInstance(GuiceContext.java:72)
	at org.locationtech.geogig.di.GuiceContext.getDecoratedInstance(GuiceContext.java:77)
	at org.locationtech.geogig.di.GuiceContext.objectDatabase(GuiceContext.java:118)
	at org.locationtech.geogig.repository.impl.RepositoryImpl.configure(RepositoryImpl.java:95)
	at org.locationtech.geogig.porcelain.InitOp._call(InitOp.java:167)
	at org.locationtech.geogig.porcelain.InitOp._call(InitOp.java:64)
	at org.locationtech.geogig.repository.AbstractGeoGigOp.call(AbstractGeoGigOp.java:154)
	at org.locationtech.geogig.cli.porcelain.Init.runInternal(Init.java:118)
	at org.locationtech.geogig.cli.AbstractCommand.run(AbstractCommand.java:68)
	at org.locationtech.geogig.cli.GeogigCLI.executeInternal(GeogigCLI.java:532)
	at org.locationtech.geogig.cli.GeogigCLI.execute(GeogigCLI.java:368)
	at org.locationtech.geogig.cli.app.CLI.run(CLI.java:95)
	at org.locationtech.geogig.cli.app.CLI.main(CLI.java:113)
Caused by: java.lang.ExceptionInInitializerError
	at org.locationtech.geogig.storage.datastream.DataStreamSerializationFactoryV2.<init>(DataStreamSerializationFactoryV2.java:52)
	at org.locationtech.geogig.storage.datastream.DataStreamSerializationFactoryV2.<clinit>(DataStreamSerializationFactoryV2.java:43)
	at org.locationtech.geogig.storage.datastream.SerializationFactoryProxy.<clinit>(SerializationFactoryProxy.java:44)
	at org.locationtech.geogig.storage.impl.AbstractObjectStore.<init>(AbstractObjectStore.java:49)
	at org.locationtech.geogig.rocksdb.RocksdbObjectStore.<init>(RocksdbObjectStore.java:95)
	at org.locationtech.geogig.rocksdb.RocksdbObjectStore.<init>(RocksdbObjectStore.java:92)
	at org.locationtech.geogig.rocksdb.RocksdbObjectDatabase.<init>(RocksdbObjectDatabase.java:51)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:488)
	at com.google.inject.internal.DefaultConstructionProxyFactory$1.newInstance(DefaultConstructionProxyFactory.java:86)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:105)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:85)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:267)
	at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:56)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:145)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1016)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103)
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1012)
	... 29 more
Caused by: java.lang.IllegalArgumentException: org.geotools.styling.StyleFactory is not an ImageIO SPI class
	at java.desktop/javax.imageio.spi.ServiceRegistry.checkClassAllowed(ServiceRegistry.java:745)
	at java.desktop/javax.imageio.spi.ServiceRegistry.<init>(ServiceRegistry.java:140)
	at org.geotools.factory.FactoryRegistry.<init>(FactoryRegistry.java:155)
	at org.geotools.factory.FactoryCreator.<init>(FactoryCreator.java:91)
	at org.geotools.factory.CommonFactoryFinder.getServiceRegistry(CommonFactoryFinder.java:76)
	at org.geotools.factory.CommonFactoryFinder.lookup(CommonFactoryFinder.java:346)
	at org.geotools.factory.CommonFactoryFinder.getFilterFactory(CommonFactoryFinder.java:300)
	at org.geotools.feature.AttributeTypeBuilder.<init>(AttributeTypeBuilder.java:192)
	at org.geotools.feature.simple.SimpleFeatureTypeBuilder.<init>(SimpleFeatureTypeBuilder.java:196)
	at org.geotools.feature.simple.SimpleFeatureTypeBuilder.<init>(SimpleFeatureTypeBuilder.java:186)
	at org.locationtech.geogig.storage.datastream.FormatCommonV2.<clinit>(FormatCommonV2.java:131)
	... 53 more
18:15:18.625 [main] ERROR o.locationtech.geogig.cli.GeogigCLI - Can't access repository at 'file:/private/tmp/geogig-test-20171022-52868-11kk5jq/.geogig'
java.lang.IllegalArgumentException: org.geotools.styling.StyleFactory is not an ImageIO SPI class
	at java.desktop/javax.imageio.spi.ServiceRegistry.checkClassAllowed(ServiceRegistry.java:745) ~[na:na]
	at java.desktop/javax.imageio.spi.ServiceRegistry.<init>(ServiceRegistry.java:140) ~[na:na]
	at org.geotools.factory.FactoryRegistry.<init>(FactoryRegistry.java:155) ~[gt-metadata-18.0.jar:na]
	at org.geotools.factory.FactoryCreator.<init>(FactoryCreator.java:91) ~[gt-metadata-18.0.jar:na]
	at org.geotools.factory.CommonFactoryFinder.getServiceRegistry(CommonFactoryFinder.java:76) ~[gt-api-18.0.jar:na]
	at org.geotools.factory.CommonFactoryFinder.lookup(CommonFactoryFinder.java:346) ~[gt-api-18.0.jar:na]
	at org.geotools.factory.CommonFactoryFinder.getFilterFactory(CommonFactoryFinder.java:300) ~[gt-api-18.0.jar:na]
	at org.geotools.feature.AttributeTypeBuilder.<init>(AttributeTypeBuilder.java:192) ~[gt-main-18.0.jar:na]
	at org.geotools.feature.simple.SimpleFeatureTypeBuilder.<init>(SimpleFeatureTypeBuilder.java:196) ~[gt-main-18.0.jar:na]
	at org.geotools.feature.simple.SimpleFeatureTypeBuilder.<init>(SimpleFeatureTypeBuilder.java:186) ~[gt-main-18.0.jar:na]
	at org.locationtech.geogig.storage.datastream.FormatCommonV2.<clinit>(FormatCommonV2.java:131) ~[geogig-core-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.storage.datastream.DataStreamSerializationFactoryV2.<init>(DataStreamSerializationFactoryV2.java:52) ~[geogig-core-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.storage.datastream.DataStreamSerializationFactoryV2.<clinit>(DataStreamSerializationFactoryV2.java:43) ~[geogig-core-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.storage.datastream.SerializationFactoryProxy.<clinit>(SerializationFactoryProxy.java:44) ~[geogig-core-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.storage.impl.AbstractObjectStore.<init>(AbstractObjectStore.java:49) ~[geogig-core-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.rocksdb.RocksdbObjectStore.<init>(RocksdbObjectStore.java:95) ~[geogig-rocksdb-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.rocksdb.RocksdbObjectStore.<init>(RocksdbObjectStore.java:92) ~[geogig-rocksdb-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.rocksdb.RocksdbObjectDatabase.<init>(RocksdbObjectDatabase.java:51) ~[geogig-rocksdb-1.2.0.jar:1.2.0]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:488) ~[na:na]
	at com.google.inject.internal.DefaultConstructionProxyFactory$1.newInstance(DefaultConstructionProxyFactory.java:86) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:105) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:85) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:267) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:56) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:145) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1016) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1012) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.spi.ProviderLookup$1.get(ProviderLookup.java:104) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.spi.ProviderLookup$1.get(ProviderLookup.java:104) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.multibindings.MapBinder$RealMapBinder$ValueProvider.get(MapBinder.java:746) ~[guice-multibindings-4.0.jar:na]
	at org.locationtech.geogig.di.FormatSelector.get(FormatSelector.java:41) ~[geogig-core-1.2.0.jar:1.2.0]
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.BoundProviderFactory.provision(BoundProviderFactory.java:72) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:61) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:62) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:145) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1016) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092) ~[guice-4.0-no_aop.jar:na]
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1012) ~[guice-4.0-no_aop.jar:na]
	at org.locationtech.geogig.di.GuiceContext.getInstance(GuiceContext.java:72) ~[geogig-core-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.di.GuiceContext.getDecoratedInstance(GuiceContext.java:77) ~[geogig-core-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.di.GuiceContext.objectDatabase(GuiceContext.java:118) ~[geogig-core-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.repository.impl.RepositoryImpl.configure(RepositoryImpl.java:95) ~[geogig-core-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.porcelain.InitOp._call(InitOp.java:167) ~[geogig-core-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.porcelain.InitOp._call(InitOp.java:64) ~[geogig-core-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.repository.AbstractGeoGigOp.call(AbstractGeoGigOp.java:154) ~[geogig-api-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.cli.porcelain.Init.runInternal(Init.java:118) ~[geogig-cli-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.cli.AbstractCommand.run(AbstractCommand.java:68) ~[geogig-cli-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.cli.GeogigCLI.executeInternal(GeogigCLI.java:532) ~[geogig-cli-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.cli.GeogigCLI.execute(GeogigCLI.java:368) ~[geogig-cli-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.cli.app.CLI.run(CLI.java:95) [geogig-cli-app-1.2.0.jar:1.2.0]
	at org.locationtech.geogig.cli.app.CLI.main(CLI.java:113) [geogig-cli-app-1.2.0.jar:1.2.0]
```

**Comment by walker** [comment]
@emerkle826 or @DBlasby, do you guys have any ideas?

**Comment by ilovezfs** [comment]
It's due to Java 9. We'll need compatibility with Java 9 for a new formula.

**Review by ilovezfs — CHANGES_REQUESTED** [review]
Needs Java 9 compatibility

**Comment by emerkle826** [comment]
GeoGig won't be able to upgrade to Java9 until GeoTools does: https://github.com/geotools/geotools/pull/1670

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@emerkle826 it looks like the PR for Java 9 compatibility was merged into geotools.

**Comment by emerkle826** [comment]
@ilovezfs The PR was merged, so GeoTools can be used with Java 9. However, it's only version 19 of GeoTools that works with Java 9, and that hasn't been officially released. Also, GeoGig 1.2.0 has been released, which includes its GeoTools 18 dependencies. GeoGig 1.2.0 _should_ be able to run with GeoTools 19 versions of those dependencies, but there is no current plan to make an official release of GeoGig 1.2.x with GeoTools 19 libraries.
I have made a PR to update GeoGig to use the snapshot versions of GeoTools 19 here: https://github.com/locationtech/geogig/pull/412, but there was an issue found in testing with java 9 (reported here: https://osgeo-org.atlassian.net/browse/GEOT-5899)

TL;DR it's going to be a while before there is an official GeoGig that runs with java 9.

**Comment by ilovezfs** [comment]
Thanks for the extended information, @emerkle826.

@walker we can reconsider this once Java 9 support is available! Thanks for the PR.
